hedgehog/src/Hedgehog/Internal/Report.hs

```diff
@@ -715,14 +718,58 @@ ppResult name (Report tests discards result) =
         ppDiscardCount discards <>
         ", passed" <+>
         ppTestCount tests <>
-        "."
+        "." <+>
+        ppClassification classes <+>
+        ppCoverage classes
 
     OK ->
       pure . icon SuccessIcon '✓' . WL.annotate SuccessHeader $
         ppName name <+>
         "passed" <+>
         ppTestCount tests <>
-        "."
+        "." <+>
+        ppClassification classes <+>
+        ppCoverage classes
+
+ppClassification :: Classification -> Doc Markup
+ppClassification (Classification classifiers total) =
+  if Map.null classifiers then
+    mempty
+  else
+    (<>) WL.linebreak $ WL.indent 4 . WL.align . WL.vsep $
+      (\(ClassifierName k, v) -> WL.text $ show (occurrenceRate v total) <> "% " <> k) <$> Map.toList classifiers
+
+occurrenceRate :: Classifier -> Integer -> Double
+occurrenceRate (Classifier _ occurrences) total =
+  let
+    percentage :: Double
+    percentage =
+      fromIntegral occurrences / fromIntegral (total - 2) * 100
+    thousandths :: Integer
+    thousandths =
+      round $ percentage * 10
+  in
+    fromIntegral thousandths / 10
+
+ppCoverage :: Classification -> Doc Markup
+ppCoverage (Classification classifiers total) =
+  let
+    coverageLines = foldMap (uncurry renderCoverage) $ Map.toList classifiers
+    renderCoverage (ClassifierName name) cl@(Classifier minRate _) =
+      if occurrenceRate cl total < minRate then
+        pure . WL.text $
+          "Only " <> show (occurrenceRate cl total) <> "% " <> name <>
+          ", but expected " <> show minRate <> "%"
+      else
+        []
```

Comment:
This isn't correct by any means and will display statistically irrelevant errors.

You should use a beta distribution as the prior of the observed class distribution and see if it's consistent (within a tolerance level) with the desired coverage.

Reply:
As a side note, using a Dirichlet prior would be the right thing to do here in the case of exclusive classifications.

But, as we can classify with overlapping cases, I believe a more "one against all" style is the best way to go.

Reply:
These results from your PR show the problem.

```
  ✓ prop_check_classifiers passed 1 test. 
      51.0% big number
      49.0% small number 
  ✓ prop_check_coverage passed 1 test. 
      51.0% big number
      49.0% small number 
  
  ⚠   Only 49.0% small number, but expected 50.0%
  ✓ 2 succeeded.
```

If we were to turn this into an error (which is kind of the point) we would hit it every time.

Reply:
This PR does not deal with turning this into a statistically relevant result. If you run X tests it will tell you the result out of those X tests.

I'd like to do what you suggest as a follow-up, because it will allow us to abort the testing if a desired coverage cannot be reached.